Dockerfile

```diff
@@ -0,0 +1,31 @@
+FROM ubuntu:xenial
+
+RUN apt-get update && apt-get install -y make build-essential libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev git netcat
+RUN curl -L https://raw.githubusercontent.com/yyuu/pyenv-installer/master/bin/pyenv-installer | bash
+
+ENV PYENV_ROOT /root/.pyenv
+ENV PATH $PYENV_ROOT/shims:$PYENV_ROOT/bin:$PATH
+
+RUN eval "$(pyenv init -)" && eval "$(pyenv virtualenv-init -)"
+RUN pyenv install 2.7.18
+RUN pyenv install 3.4.10
+RUN pyenv install 3.5.9
+RUN pyenv install 3.6.11
+RUN pyenv install 3.7.8
+RUN pyenv install 3.8.5
+RUN pyenv install pypy2.7-7.1.1
+RUN pyenv local 2.7.18 3.4.10 3.5.9 3.6.11 3.7.8 3.8.5 pypy2.7-7.1.1
+
+ENV MONGODB=4.1.1
+RUN wget http://fastdl.mongodb.org/linux/mongodb-linux-x86_64-$MONGODB.tgz
+RUN tar xzf mongodb-linux-x86_64-$MONGODB.tgz
+RUN mongodb-linux-x86_64-$MONGODB/bin/mongod --version
+RUN mkdir -p /data/db
+RUN ln -s /mongodb-linux-x86_64-$MONGODB/bin/mongod /usr/bin/mongod
+
+RUN pip install tox
+
+ADD . mongomock
```

Comment:
Note that adding . will end up adding a lot of things inside the container. I feel that it's ok when mounting it as a volume, but the copy here might be long. I suggest you cherry-pick strictly the files and folders that are required, or even drop the files entirely and force the use of volumes mounting.

In both cases, I suggest that you add a `.dockerignore` file to make sure the build step doesn't prepare all files in here to be copied (for instance the `.git` folder might be big).

Reply:
:100: to this

The volume mounting in vanilla Dockerfile is...less than appealing, for what I'd like to do. This is actually precisely why I was mentioning docker-compose originally - it's volume mounting is great and would remove the need to do a `docker build` every time you make a change.

However, I'm still going to hold off on the docker-compose changes for a few reasons:
* I want to make sure the vanilla Docker content is in a solid place
* Following that - some folks may not be using docker-compose, so I don't want to pigeon hole anyone into one way to do this

TL;DR - the `.dockerignore` is a must have.

Reply:
Annnnd I think I've hit a snag with isolating this content from the container...

> `Exception: Versioning for this project requires either an sdist tarball, or access to an upstream git repository. `

Without the `.git` directory - tox fails on the `GLOB sdist-make: /mongomock/setup.py` with the above...unless I add it back.

My sketch for `docker-compose` wouldn't have this problem as:
* Dockerfile would only be lines 1-17 + `pip install tox`
* docker-compose.yml would handle mongo image to use
* docker-compose.yml can cleanly mount host folder to container FS...no copy/add/etc